Sources/PackageLoading/TargetSourcesBuilder.swift

```diff
@@ -201,8 +201,11 @@ public struct TargetSourcesBuilder {
 
         // Match any sources explicitly declared in the manifest file.
         if let declaredSources = self.declaredSources {
+            var pathIsDeclared = false
+            let pathShouldBeDeclared = rules.filter { $0.rule == .compile }.contains(where: { $0.match(path: path, toolsVersion: toolsVersion) })
```

Comment:
Actually, does this apply only to compilation?  What about resources?

Reply:
It's not needed for resources as there's no initialized resources rules (for this target) to check against; not sure why no such rule was added at initialization, unlike source rules (compile). Any undeclared resources at this point seem to be handled by another resource validation at a later time.

Reply:
That seems surprising.  I have work to do in this area in any case to make sure we don't emit unhandled-files warnings for files that are handled by built-tool plugins, so I will take a closer look and see if I can offer some insights as to what's going on (I'm not that familiar with this part of the code base).